src/AggregateFunctions/AggregateFunctionRangeSum.cpp

```diff
@@ -0,0 +1,56 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/AggregateFunctionRangeSum.h>
+#include <AggregateFunctions/FactoryHelpers.h>
+#include <AggregateFunctions/Helpers.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+
+#include <ext/range.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+    template <template <typename> class Data>
+    AggregateFunctionPtr createAggregateFunctionRangeSum(const std::string & name, const DataTypes & arguments, const Array &)
+    {
+        if (arguments.size() != 2)
+            throw Exception(
+                "Aggregate function " + name + " requires two timestamps argument.", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        for (const auto i : ext::range(1, arguments.size()))
+        {
+            const auto * arg = arguments[i].get();
+            if (!WhichDataType(arg).isNativeUInt())
```

Comment:
I suppose it can work with signed too, or even with any numeric, including floats or date/time?

Reply:
The return type is `UInt64`, we need to make sure that it has overload for plus with `UInt64` and DataType of `start` and `end`, for example, if `start` has DataType `UInt128` and `end` has DataType `Float`, it doesn't work. Or we should constrain that `start` and `end` should have same DataType?

Reply:
Yes, I think `start` and `end` should have same type.